088-rebalance-progress-status.md

```diff
@@ -0,0 +1,421 @@
+
+[4] The broker load from the optimization proposal as a JSON string that already maintained in the `ConfigMap`.
+
+### Field values per `KafkaRebalance` State
+
+We will provide the progress information for the `KafkaRebalance` states where it is relevant:
+
+- `ProposalReady`:
+  - `estimatedTimeToCompletion`: The theoretical minimum estimated time it would take the rebalance to complete if executed.
+  - `completedDataMovementPercentage`: 0
+  - `executorState`: Empty JSON object.
+- `Rebalancing`
+  - `estimatedTimeToCompletion`: The estimated time it will take for the ongoing rebalance to complete.
+  - `completedDataMovementPercentage`: The percentage of data movement of the ongoing partition rebalance that is complete.
+  - `executorState`: JSON object from [/kafkacruisecontrol/state?substates=executor](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-state-of-cruise-control) endpoint of Cruise Control REST API.
+- `Stopped`
+  - `estimatedTimeToCompletion`: N/A
+  - `completedDataMovementPercentage`: The value of `completedDataMovementPercentage` from previous update.
+  - `executorState`: JSON object from previous update.
+- `NotReady`
+  - `estimatedTimeToCompletion`: N/A
+  - `completedDataMovementPercentage`: The value of `completedDataMovementPercentage` from previous update.
+  - `executorState`: JSON object from previous update.
+- `Ready`:
+  - `estimatedTimeToCompletion`: 0
+  - `completedDataMovementPercentage`: 100
+  - `executorState`: Empty JSON object.
+
+All the information required for the Cluster Operator to estimate the values of `estimatedTimeToCompletionInMinutes` and `completedDataMovementPercentage` fields can be derived from the Cruise Control server configurations and the [/kafkacruisecontrol/state?substates=executor](https://github.com/linkedin/cruise-control/wiki/REST-APIs#query-the-state-of-cruise-control) REST API endpoint.
+However, the actual formulas used to produce values for these fields depend on the state of the `KafkaRebalance` resource.
+Checkout the example in the [Field: executorState](#field-executorstate) section to see where the fields used in the formulas below come from.
+
+### Field: `estimatedTimeToCompletionInMinutes`
+
+The estimated time it will take in minutes for a rebalance to complete.
+
+The formulas used to calculate the field value differ for each applicable `KafkaRebalance` state:
+
+#### State: `ProposalReady`
+
+**Estimation for inter-broker rebalance:**
+
+This estimate will be a theoretical minimum derived from Cruise Control capacity and throttle configurations.
+This means that the cluster rebalance would take at least the estimated amount of time to complete.
+
+$$\text{maxPartitionMovements}_{[1]} = \min(\text{numberOfBrokers} \times \text{num.concurrent.partition.movements.per.broker}),\text{max.num.cluster.partition.movements})$$
```

Comment:
I can't follow this calculation, is it calculating the maximum number of non-concurrent partition movements? I can't work out why the number of brokers is needed. Isn't the worst case scenario that all movements have to happen from a single broker, so should it be `max.num.cluster.partition.movements/num.concurrent.partition.movements.per.broker`? Also it looks like you are missing a bracket somewhere, possible at the beginning of `numberOfBrokers x num.concurrent.partition.movements.per.broker`?

Reply:
@katheris There were unnecessary calculations in the originally provided formula, they have been fixed/simplified. This should now be addressed!